visualizer/ui.js

```diff
@@ -120,7 +124,15 @@ class Ui extends events.EventEmitter {
   }
 
   selectHottestNode (opts) {
-    this.selectNode(this.dataTree.getFrameByRank(0), opts)
+    const node = this.dataTree.getFrameByRank(0)
+    const nodeInvalidMessage = ' node selected in selectHottestNode'
+
+    // Prevent infinite loop if some future bug allows an invalid node to be returned here
+    if (!node) throw new Error ('No' + nodeInvalidMessage)
```

Comment:
Little heads-up: You might have a Lint error here because of the space between `Error` and `('`

Reply:
Strangely, the linter doesn't mind, but it probably _should_ mind, so I've changed it anyway 🙃